### DISCUSSION TITLE
Is possible show the Torus Confirmation pop-up for a transaction with Solidity?

### DISCUSSION THREAD

**Discussion by pr0mming** [Web3Auth/Web3Auth#789] [Help]
Hello there,

I am new with Web3Auth, Ether.js and Solidity, I'm doing an application where I need buy and sell NFT items, you can know I have a Smart Contract for manage this actions with my tokens, and the interaction between the Web and its is with Ether.js.

I don't know but when you have a Metamask wallet (browser extension) and call a Smart Contract method for pay for the NFT (payable) the Metamask extension come up in my browser and ask you for confirm the transaction (NFT price + Gas) otherwise you aren't able to continue the flow.

In my case I have the Web3Auth provider and I found the [Torus Wallet Connector Plugin](https://web3auth.io/docs/sdk/web/plugins/torus-wallet) and it show in my app for interact with my torus wallet, but even though my app use this plugin this doesn't show any confirmation pop-up, of course the crypto money is charged successfully, but I would like confirm the transaction before like Metamask.

I want this but with Torus, It would be possible? or maybe exists a large difference between Metamask and Torus, I mean, for be a browser extension, Is it the reason for Metamask can dispatch the pop-up? or Do I need do it this from scratch (I mean, design the screen, calculate the gas price for show, etc.)?

I am really sorry if this is a dumb question but I am new in this field, and I haven't found usefully info about this :c

**Comment by shahbaz17** [comment]
Hey @pr0mming 

If you're to use TorusWallet to log in from this screen, 
<img width="394" alt="Screenshot 2022-10-31 at 1 51 36 PM" src="https://user-images.githubusercontent.com/6962565/198963778-6614765a-d4fe-4a52-81a2-e4531c6b05be.png">

And when making a `sendTransaction`, `signTransaction` or `signMessage`, the pop-up would open like this:
<img width="1104" alt="Screenshot 2022-10-31 at 1 51 53 PM" src="https://user-images.githubusercontent.com/6962565/198963828-a2b488e3-8737-4e41-ac33-f68a968dc0c3.png">

**Reply by pr0mming** [reply]
Thank you very much for your answer, but in my case is different, I am trying to call a Smart Contract method, like this with ether.js using the web3Auth provider:

```
const provider = new ethers.providers.Web3Provider(web3AuthProvider)
const signer = provider.getSigner()

const abiData = JSON.parse(abiDefinition)
const contract = new ethers.Contract(contractAddress, abiData, signer)

const salePrice = ethers.utils.parseUnits('50', 'ether')

// makeNFTSale is my method
const transaction = await contract.makeNFTSale(tokenId, { value: salePrice })
await transaction.wait()
```
You can see I am not using any method like `sendTransaction`, `signTransaction` or `signMessage` because since my knowledge (I can be wrong), this methods is for send cash to other address, right? The most strange of my implementation is that works fine, but the pop-up never is showed :thinking:

**Reply by shahbaz17** [reply]
Hey @pr0mming 

If you're performing a write operation on your smart contract, it should prompt you to confirm.

And just to confirm, you're using Torus Wallet from Web3Auth Modal?

**Reply by pr0mming** [reply]
@shahbaz17 This is my implementation, but basically I am using TorusWallerConnectorPlugin and I tried add the TorusWalletAdapter without luck

I use the web3Auth provider instance for pass into the Web3Provider Ethers.js method (the snippet code I put before)

Maybe I am using something wrong in my code?
![image](https://user-images.githubusercontent.com/20020612/199250961-8d84970b-57b8-4f43-889a-6ccb8595085f.png)

**Comment by shahbaz17** [comment]
![Screenshot 2022-11-02 at 9 08 32 AM](https://user-images.githubusercontent.com/6962565/199389841-8ada064a-5066-4ea1-b0b7-e393a7b2caf8.png)

May I know why you are adding both? 

And it looks like you're using `@web3auth/modal` package. So, On the Modal, TorusWallet is not shown inside the external wallet?

![Modal](https://user-images.githubusercontent.com/6962565/198963778-6614765a-d4fe-4a52-81a2-e4531c6b05be.png)

**Reply by pr0mming** [reply]
Yeah, TorusWalletConnectorPlugin is for show the Torus extension in the left-bottom ubication of my front-end app:

![image](https://user-images.githubusercontent.com/20020612/199593805-a9ed79b3-dfe1-415c-8975-764a7407811b.png)

And TorusWalletAdapter was a silly and desperate try for show the pop-up...

But I have figure out about the "Wallet Connection" control is below the other SignIn methods, sorry :cry:

![image](https://user-images.githubusercontent.com/20020612/199596087-205a2a79-3690-4284-9fc4-47187b93db55.png)

![image](https://user-images.githubusercontent.com/20020612/199595972-54b794b3-360f-4cc0-b364-d04919b26d27.png)

And now I can see the pop-up!

![image](https://user-images.githubusercontent.com/20020612/199597230-bdfc7067-de61-4ef6-8673-a2bc9768400f.png)

Works perfectly but I have a question, Why this doesn't work with the other SignIn methods? I mean, if you see the second picture of this comment, exists the Social networks for login, if I login with Google or Facebook for example, and try to do the same action, the Torus pop-up doesn't appears, this is due because I need stay connected with the "Wallet Connection" method instead the others ?

**Reply by shahbaz17** [reply]
This is an expected behavior if signing through socials. To make the Web3 experience more seamless and behind the scene.

If your use case requires a popup for each transaction. Then I would recommend to hide all other options on Web3Auth Modal or use [TorusEmbed](https://github.com/torusresearch/torus-embed).

**Reply by pr0mming** [reply]
Hmm ok, thanks a lot for your help and patient, Cheers!
